### ISSUE TITLE
Have you finetune the center features?

### ISSUE THREAD

**Issue by globallocal** [happynear/NormFace#1]
Hello, Mr. Wang, 
        I read your paper today. I'm interesting with the fine-tune process.  What' the fine-tune iterations in the experiments, do all the experiments have the same iterations, and have you fine-tune the center features at the same time? Thank you so much. Looking forward for your reply.

**Comment by happynear** [comment]
Using a batch size of 256, we only need about 5000 iterations to get the network converge. The learning rate is 0.001. The weights of the newly added layers are initialized to be gaussian noise. The performance of center loss is also listed in the paper.

BTW, I re-implement the center loss by setting the margin of general_contrastive_loss to 0.

**Comment by globallocal** [comment]
Than you so much, have you ever try to train a no fine-tune model, and how is the performance?

**Comment by happynear** [comment]
Training the scaled softmax from scratch is similar with softmax.
Training the C-contrastive loss from scratch, we can get 98.9%, which is very close to center loss, but not beyond it.

**Comment by globallocal** [comment]
Good performance, I'will pay more attention to the method, thank you again.

**Comment by PierreHao** [comment]
I have read the your Mirror Face (Flip face). I have some experience about it. Sometimes it works, sometimes it does not work as you mentioned. If we train it with flip, it will have more chance work. I think sometimes it does not work, due to the too simple feature ensemble method, we do not take advantage of the information of each feature, for example the incorrelation of the two feature etc.  @happynear

**Comment by happynear** [comment]
@PierreHao ,
May we open a new issue to talk about it?

**Comment by PierreHao** [comment]
No problem, let me read your paper first, then we could discuss some problem. Does your Norm Face has a relationship with SphereFace?

**Comment by happynear** [comment]
I and Yandong often talk online. But he doesn't want to give details about his SphereFace. So let's wait for his work being public available.

There is one thing I know that SphereFace gets >99.4% on LFW with only Webface as training data.

**Comment by PierreHao** [comment]
Yes, the result of SphereFace is amazing. I have tried my best with Webface, the result is only 99.1%

**Comment by globallocal** [comment]
Hello, Mr. Wang:
        I have an another question. The paper "NormFace" use the 10-fold evaluation, how is the standard error of the mean Se ? Thank you.

**Comment by globallocal** [comment]
Hello, Mr. Wang:
        I carefully read your paper yesterday. I really think the proofs help me understand Softmax loss and NormFace method better. 
        At the same time, I'm so confused about the number of the training faces of WebFace. Since I use less than 0.46M detected face images  to train ResNet (Wen's model), the accuracy never achieves about 99%. Is it reasonable?  And what's the number of your training face images? If the number is about 0.49M, how do you achieve the number? 
       Looking forward to your reply. Thank you so much.

**Comment by happynear** [comment]
@globallocal , have you achieved the performance of the author's released model?

**Comment by globallocal** [comment]
Under no fine-tune condition, never achieve about 99%. I train the released model using center loss with my less than 0.46 WebFace database, achieved less than 98%.

**Comment by globallocal** [comment]
Though the performance is not so good, I think it's my real results, and I will stick to it. Thank you again for answer.

**Comment by PierreHao** [comment]
try to change center loss loss_weight, and when you do test, do flip, subtract mean feature, check your aligned lfw faces because in some images there are more than one person.

**Comment by globallocal** [comment]
Thank you so much, I will check the first and last suggests.

**Comment by happynear** [comment]
@globallocal , actually I followed all the settings from [Yandong's repo](https://github.com/ydwen/caffe-face/tree/caffe-face/face_example). I got ~99% at the first try.

You may use [my alignment code](https://github.com/happynear/FaceVerification/blob/master/dataset/general_align.m) . Another thing is the batchsize, you may use `iter_size` parameter in `solver.prototxt` to simulate big batchsize. 

There is no more magic. I believe you can also get a good result.

**Comment by globallocal** [comment]
@happynear Thank you so much.
